lib/types.d.ts

```diff
@@ -0,0 +1,52 @@
+declare module 'json5' {
+  type StringifyOptions = Partial<{
+    /**
+     * A function that alters the behavior of the stringification process, or an array of String and Number objects that serve as a whitelist for selecting/filtering the properties of the value object to be included in the JSON5 string. If this value is null or not provided, all properties of the object are included in the resulting JSON5 string.
+     */
+    replacer: (this: any, key: string, value: any) => any;
```

Comment:
`replacer` can also be an array of string and numbers.

Reply:
The JSON definition uses the following overloads:

```ts
stringify(value: any, replacer?: (this: any, key: string, value: any) => any, space?: string | number): string
stringify(value: any, replacer?: (string | number)[], space?: string | number): string
```

However, we have a third one:

```ts
stringify(value: any, options: StringifyOptions): string
```

where `StringifyOpions` also has a `replacer` property which can either be a function or array of string and numbers.

I'm not sure what the best way is to define this. Maybe we have a `Replacer` type that is a union of `ReplacerFunction` and `ReplacerArray`?

Reply:
I have `(this: any, key: string, value: any) => any | Array<string | number>` for all instances of `replacer` in the definition file now if that's ok. Unless I'm misunderstanding something...

Reply:
I was just thinking about following the DRY principle, but I think declaring the same signature in multiple places is okay here.